src/modules/powerrename/dll/dllmain.cpp

```diff
@@ -0,0 +1,252 @@
+        {
+            DllAddRef();
+        }
+        else
+        {
+            DllRelease();
+        }
+        return S_OK;
+    }
+
+private:
+    ~CSmartRenameClassFactory()
+    {
+        DllRelease();
+    }
+
+    long m_refCount;
+    CLSID m_clsid;
+};
+
+BOOL WINAPI DllMain(HINSTANCE hInstance, DWORD dwReason, void*)
+{
+    switch (dwReason)
+    {
+    case DLL_PROCESS_ATTACH:
+        g_hInst = hInstance;
+        break;
+
+    case DLL_PROCESS_DETACH:
+        break;
+    }
+    return TRUE;
+}
+
+//
+// Checks if there are any external references to this module
+//
+STDAPI DllCanUnloadNow(void)
+{
+    return (g_dwModuleRefCount == 0) ? S_OK : S_FALSE;
+}
+
+//
+// DLL export for creating COM objects
+//
+STDAPI DllGetClassObject(_In_ REFCLSID clsid, _In_ REFIID riid, _Outptr_ void **ppv)
+{
+    *ppv = NULL;
+    HRESULT hr = E_OUTOFMEMORY;
+    CSmartRenameClassFactory *pClassFactory = new CSmartRenameClassFactory(clsid);
+    if (pClassFactory)
+    {
+        hr = pClassFactory->QueryInterface(riid, ppv);
+        pClassFactory->Release();
+    }
+    return hr;
+}
+
+STDAPI DllRegisterServer()
+{
+    return S_OK;
+}
+
+STDAPI DllUnregisterServer()
+{
+    return S_OK;
+}
+
+void DllAddRef()
+{
+    g_dwModuleRefCount++;
+}
+
+void DllRelease()
+{
+    g_dwModuleRefCount--;
+}
+
+
+class PowerRenameModule : public PowertoyModuleIface
+{
+private:
+    // Enabled by default
+    bool m_enabled = true;
+
+public:
+    // Return the display name of the powertoy, this will be cached
+    virtual PCWSTR get_name() override
+    {
+        return L"PowerRename";
+    }
+
+    // Enable the powertoy
+    virtual void enable()
+    {
+        m_enabled = true;
+        save_settings();
+    }
+
+    // Disable the powertoy
+    virtual void disable()
+    {
+        m_enabled = false;
+        save_settings();
+    }
+
+    // Returns if the powertoy is enabled
+    virtual bool is_enabled() override
+    {
+        return m_enabled;
+    }
+
+    // Return array of the names of all events that this powertoy listens for, with
+    // nullptr as the last element of the array. Nullptr can also be retured for empty list.
+    virtual PCWSTR* get_events() override
+    {
+        return nullptr;
+    }
+
+    // Return JSON with the configuration options.
+    // These are the settings shown on the settings page along with their current values.
+    virtual bool get_config(_Out_ PWSTR buffer, _Out_ int* buffer_size) override
+    {
+        HINSTANCE hinstance = reinterpret_cast<HINSTANCE>(&__ImageBase);
+
+        // Create a Settings object.
+        PowerToysSettings::Settings settings(hinstance, get_name());
+        settings.set_description(L"A Windows Shell Extension for more advanced bulk renaming using search and replace or regular expressions.");
+
+        // Link to the GitHub PowerRename sub-page
+        settings.set_overview_link(L"https://github.com/microsoft/PowerToys/tree/master/src/modules/PowerRename");

```

Comment:
Looks like GitHub is case sensitive. It should be `https://github.com/microsoft/PowerToys/tree/master/src/modules/powerrename`